src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const stateArr = [stateCopy];
+
+  for (const action of actions) {
+    let currentState = { ...stateArr[stateArr.length - 1] };
+
+    switch (action.type) {
+      case 'addProperties' :
+        currentState = Object.assign({}, currentState, action.extraData);
+        break;
+
+      case 'removeProperties' :
+        for (const key of action.keysToRemove) {
+          delete currentState[key];
+        }
+        break;
+
+      case 'clear' :
+        currentState = {};
+        break;
+
+      default: {
+        break;
+      }
+    }
+
+    stateArr.push(currentState);
+  }
+
+  stateArr.shift();
```

Comment:
it can work without shift if you set initial value of the `stateArr` as `[]`

Reply:
don't working